schemas/alpha/azure.yaml.json

```diff
@@ -161,6 +161,16 @@
                                 "description": "Runs after the service dependencies are restored",
                                 "$ref": "#/definitions/hook"
                             },
+                            "prebuild": {
```

Comment:
The only thing I am concerned about is that `azd build` is not documented (currently hidden). Should we consider making `azd build` public?

Reply:
This has nothing to do with the `azd build` command.  It is just a lifecycle event that the user can integrate with during the lifecycle evens within a service.

Additionally we could also choose to unhide the `build` command.

Reply:
I might be looking at it different here. 
In my mind, hooks are directly related to azd-commands b/c that's the application API's.
The lifecycle is an implementation detail behind the azd-commands. 

As a customer, if I read the output of `azd help` or go to the official [azd documentation for hooks](https://learn.microsoft.com/azure/developer/azure-developer-cli/azd-extensibility) I can see:

> The hooks middleware allows you to execute custom scripts before and after azd commands and service lifecycle events

I can learn about `azd commands` but how can I know what's the `service's lifecycle events` ?  

The way I was thinking and looking at `service's lifecycle events` is as one `azd command that is scoped to one service only`.  That way, I can self-discover what is the lifecycle of a service. Otherwise, how do I know what are the steps of the lifecycle for my service? where do I find/learn about it?

So, my concern is still about the complexity about the official documentation, the schema for azure.yaml and understanding what are those lifecycle events.

Reply:
@vhvb1989 The [docs](https://github.com/Azure/azure-dev/pull/4069) call out the current lifecycle events.  Lifecycle events are fired on services dependent on which `azd` commands are run. 

An example - if you call `azd deploy` both the `package` and `deploy` service events are raised and will call any registered hooks for those services.  Sometimes they are 1:1 between the azd command and hook, sometimes additional lifecycle events will also be fired based on dependent lifecycle calls.  For example, before you can deploy a service it needs to be built and packaged unless you are deploying from an already created package.

Reply:
Docs have been updated to reflect `prebuild` and `postbuild`.